### PR TITLE
[CARBONDATA-1608]Support Column Comment for Create Table

### PR DESCRIPTION
**Pull request by akashrn5**
add column comment during carbon create table and when table is described if comment is not mentioned, default comment will be null

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [*] Make sure the PR title is formatted like:
   `[CARBONDATA-<Jira issue #>] Description of pull request`
   
 - [*] Make sure to add PR description including
      -create carbon table with column comment is added in this PR, previously there was no option for column comment, now it has been added
-for implementing this, the parser has been edited to get column comment and instead of checking its encoding technique to display in desc sql, we can just show the comment for that specific column if any, if there are no comment for the columns, the default value is null  
        - the root cause/problem statement
        - What is the implemented solution

 - [*] Any interfaces changed?
 NONE
 - [*] Any backward compatibility impacted?
 NONE
 - [*] Document update required?
NONE
 - [*] Testing done
 
test cases are added for this 
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
         
 - [*] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
                 
---

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1300/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1301/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/664/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1302/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/665/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/673/

**Comment by akashrn5** [comment]
@manishgupta88 please review

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1310/

**Comment by akashrn5** [comment]
@jackylk please review

**Comment by akashrn5** [comment]
@gvramana please review

**Comment by akashrn5** [comment]
@ravipesala please review

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala [hunk 1]
```diff
@@ -424,11 +424,20 @@ class CarbonSpark2SqlParser extends CarbonDDLSqlParser {
...
   def getFields(schema: Seq[StructField]): Seq[Field] = {
     schema.map { col =>
-      val x = if (col.dataType.catalogString == "float") {
-        '`' + col.name + '`' + " double"
-      }
-      else {
-        '`' + col.name + '`' + ' ' + col.dataType.catalogString
+      val colComment = col.getComment()
+      val x = if (colComment.isDefined) {
+        val columnComment = " comment \"" + colComment.get + "\""
```
Comment (gvramana) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala]: No need to repeat logic, just add columnComment as empty string if no colComment

File: integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesInsertTest.scala [hunk 2]
```diff
@@ -945,4 +948,41 @@ class BooleanDataTypesInsertTest extends QueryTest with BeforeAndAfterEach with
     )
   }
 
+  test("Inserting table with bad records, and SORT_COLUMNS is boolean column") {
+    CarbonProperties.getInstance().addProperty(CarbonCommonConstants.ENABLE_AUTO_LOAD_MERGE, "true")
```
Comment (gvramana) [inline integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesInsertTest.scala]: How is this test related to comments?

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala [hunk 3]
```diff
@@ -65,6 +65,7 @@ private[sql] case class CarbonDescribeFormattedCommand(
     val dims = relation.metaData.dims.map(x => x.toLowerCase)
     var results: Seq[(String, String, String)] = child.schema.fields.map { field =>
       val fieldName = field.name.toLowerCase
+      val colComment = field.getComment().getOrElse("null")
```
Comment (gvramana) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala]: Also needs to make sure that "desc table" also works. As spark/Hive supports accordingly, need to make it consistant

File: integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesInsertTest.scala (hunk 2 above)
Comment (akashrn5) [reply integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesInsertTest.scala]: this test is not related to col comment, this test is just added in this PR which is for boolean datatype

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala (hunk 1 above)
Comment (akashrn5) [reply integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala]: handled, please review

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala (hunk 3 above)
Comment (akashrn5) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala]: verified, it is in sync with hive, if comment is not there, by default it will be null

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1171/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1748/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1176/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1752/
